src/main/java/core/basesyntax/GetFigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class GetFigureSupplier extends GetFigure {
+    private static final int RANGE = 100;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static int index;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private String figureType;
+    private GetFigure figure;
+    private String figureColor;
+
+    public GetFigure getRandomFigure() {
+        index = random.nextInt(FigureType.values().length);
+        figureType = FigureType.values()[index].name();
+        figureColor = colorSupplier.getRandomColor();
+        switch (figureType) {
+            case "SQUARE":
+                figure = new Square(random.nextInt(RANGE));
+                figure.setColor(figureColor);
+                return figure;
```

Comment:
extract logic from each case to dedicated private method

Reply:
Disagree... I don't  understandt why?